felix/ipsets/ipsets.go

```diff
@@ -462,16 +475,28 @@ func (s *IPSets) tryResync() (err error) {
 		if debug {
 			s.logCxt.Debugf("Parsing IP set %v.", name)
 		}
-		err = s.resyncIPSet(name)
-		if err != nil {
-			s.logCxt.WithError(err).Errorf("Failed to parse ipset %v", name)
-			return
+		if err = s.resyncIPSet(name); err != nil {
+			// Ignore failures of IP sets not in desired state, as those will be cleaned up later.
+			_, desired := s.setNameToProgrammedMetadata.Desired().Get(name)
+			if desired {
+				failedIPSets = append(failedIPSets, name)
+			}
+			if debug {
+				s.logCxt.WithError(err).WithFields(log.Fields{
+					"name":    name,
+					"desired": desired,
+				}).Error("Failed to parse an ipset")
+			}
```

Comment:
I think it's worth logging at warning level here (and worth tailoring the message to avoid worrying users immediately.
```suggestion
			if desired {
				s.logCxt.WithError(err).WithFields(log.Fields{
					"name":    name,
				}).Warn("Failed to parse required Calico-owned ipset that is needed, will try recreating it.")
			} else {
				s.logCxt.WithError(err).WithFields(log.Fields{
					"name":    name,
				}).Warn("Failed to parse Calico-owned ipset that is no longer needed, will queue it for deletion.")
			}
```